### DISCUSSION TITLE
creating arrow files for species of large genome

### DISCUSSION THREAD

**Discussion by Fuyt27** [GreenleafLab/ArchR#1681] [Questions / Documentation]
I want to use ArchR to study data of axolotl.
but when creating arrow files, I found that the tbi index of my fragments cannot be built since axolotl's genome is too large.
So how to do with it?
Looking forward to your early reply

**Comment by rcorces** [comment]
I guess you are bumping up against this limit:
http://www.htslib.org/doc/tabix.html#:~:text=The%20tabix%20(.tbi)%20and%20BAI%20index%20formats%20can%20handle%20individual%20chromosomes%20up%20to%20512%20Mbp%20(2%5E29%20bases)%20in%20length.%20If%20your%20input%20file%20might%20contain%20data%20lines%20with%20begin%20or%20end%20positions%20greater%20than%20that%2C%20you%20will%20need%20to%20use%20a%20CSI%20index.

You could use bam files as input to ArchR and bypass tabix. I'm in the process of updating the documentation on how to use bam files and have pasted that updated documentation below as it isnt on the website yet.

```
## Using BAM files for Arrow File creation

Though fragment files are strongly preferred because of their very standardized format, BAM files can also be used as input for Arrow File creation in ArchR. The key to getting this to work is understanding your BAM file structure, in particular which field of the BAM file contains the cell-specific barcode information. In this example, we will download some data from 10x Genomics to illustrate how to use BAM files as input to `createArrowFiles()`. Using `samtools view` 

dir.create(path = "./PBMC_BAM")
download.file(url = "https://cf.10xgenomics.com/samples/cell-atac/2.0.0/atac_pbmc_500_nextgem/atac_pbmc_500_nextgem_possorted_bam.bam",
              destfile = "./PBMC_BAM/atac_pbmc_500.bam")
download.file(url = "https://cf.10xgenomics.com/samples/cell-atac/2.0.0/atac_pbmc_500_nextgem/atac_pbmc_500_nextgem_possorted_bam.bam.bai",
              destfile = "./PBMC_BAM/atac_pbmc_500.bam.bai")
stopifnot(tools::md5sum("./PBMC_BAM/atac_pbmc_500.bam") == "8140a2218ecdfd276aca5c4bb999c989")
stopifnot(tools::md5sum("./PBMC_BAM/atac_pbmc_500.bam.bai") == "d3c5f5a00fec76378f2a947749ff2cf5")

If we were to use `samtools view` from the command line to look at this BAM file, the first line would look like this:

A00519:269:H7FM2DRXX:2:1201:1985:18834  83      chr1    9997    0       50M     =       10022   -25     CAGATAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC  ,,,:FF,,F,FF,FFFFFFF:FFFFFFFFFFFFFFF:FF:FFF,:FFFFF      NM:i:1  MD:Z:1C48       AS:i:48 XS:i:47 CR:Z:GCGGGTTAGAACGTCG       CY:Z:FFFFFFFF:FFFFFFF   CB:Z:GCGGGTTAGAACGTCG-1 BC:Z:ATCGTACT   QT:Z:FFFFFFFF   RG:Z:atac_pbmc_500_nextgem:MissingLibrary:1:H7FM2DRXX:2

Here, the `CB` tag is used to store the cell-specific barcode, in this case `CB:Z:GCGGGTTAGAACGTCG-1`. This means that we will pass `bcTag = "CB"` to `createArrowFiles()` to tell it to look for the `CB` tag. If, for example, your BAM files come from a scATAC-seq technology other than 10x Genomics, the relevant tag will likely be different. On top of that, the format of the cell-specific barcode may also be different. In these cases, it may be necessary to use the `gsubExpression` parameter to clean up the cell-specific barcode string. The other important input parameter is `bamFlag` which will determine which reads or fragments are viewed as valid. The value passed to `bamFlag` should be in the format of a `scanBamFlag` used by `ScanBam` in `Rsamtools`.

addArchRGenome("hg38")

ArrowBam <- createArrowFiles(
  inputFiles = "./PBMC_BAM/atac_pbmc_500.bam",
  sampleNames = "bam10x",
  minTSS = 4, #Dont set this too high because you can always increase later
  minFrags = 1000, 
  addTileMat = FALSE,
  addGeneScoreMat = FALSE,
  bcTag = "CB",
  bamFlag = list(isMinusStrand = FALSE, isProperPair = TRUE, isDuplicate = FALSE),
  force = TRUE
)
```

**Reply by Fuyt27** [reply]
Thanks for your reply! But I am afraid it won't work for me since  if I perform 'samtools index' to my bam file, it turns out that the bam is also too large and here is the error info:
Region 1132363227..1132363282 cannot be stored in a bai index. Try using a csi index with min_shift = 14, n_lvls >= 6
samtools index: failed to create index for "Aligned.out.bam": Numerical result out of range.

So how to do with it? I would appreciate it if you offer a solution.

**Reply by rcorces** [reply]
Sorry. If you cannot create a fragments file or a bam file, then ArchR is not the tool for you. This is (currently unfunded) academic software and we dont have the bandwidth to address such issues that effect very small portions of the potential user base.

**Reply by Fuyt27** [reply]
Thanks for your reply anyway! ArchR is excellent at scATAC-seq analysis so how wish i could use it to work on axolotl data.
